src/java/picard/vcf/CallingMetricAccumulator.java

```diff
@@ -90,17 +109,19 @@ public CallingMetricAccumulator(final DbSnpBitSetUtil.DbSnpBitSets dbsnp) {
     /** Incorporates the provided variant's data into the metric analysis. */
     @Override
     public void accumulate(final VariantContext vc) {
-        progress.record(vc.getChr(), vc.getStart());
+        progress.record(vc.getContig(), vc.getStart());
         if (!isVariantExcluded(vc)) {
             final String singletonSample = getSingletonSample(vc);
             updateSummaryMetric(summaryMetric, null, vc, singletonSample != null); // The summary metric has no genotype.
-            for (final String sampleName : vc.getSampleNames()) {
-                // Skip homozygous reference calls.
-                if (!vc.getGenotype(sampleName).isHomRef()) {
-                    updateDetailMetric(sampleMetricsMap.get(sampleName), vc.getGenotype(sampleName), vc,
-                            sampleName.equals(singletonSample));
-                }
-            }
+
+            // Skip homozygous reference calls.
+            vc.getSampleNames()
```

Comment:
It's me that's missing something.  I didn't realize that GenotypesContext implements List, which has a default implementation of `stream()`.